### PR TITLE
feat: showing the user roles in messages to achieve the original RocketChat UI

### PR DESCRIPTION
**Pull request by Rec0iL99**
# Showing the user roles in messages to achieve the original RocketChat UI

## Acceptance Criteria fulfillment

- [x] Calling the `/channels.roles` endpoint to get all the roles in a channel
- [x] Storing all the roles in a zustand store
- [x] Displaying the correct roles for a user using `username` as an identifier
- [x] Used fuselage's `Message.Role` component to display the role
- [x] Not showing anything when no role is present for a user

## Video/Screenshots

Screenshot 2023-01-30 at 11 53 49 PM

### CODE REVIEW

**Comment by Rec0iL99** [comment]
@sidmohanty11 would love your review on this. Thanks!

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]
Thanks @Rec0iL99 ,  I definitely think it's a great nice to have feature to show the roles of users. I am just worried about how it will affect our API usage and the first waterfall ( we want to keep EmbeddedChat as light as possible). I am thinking can we change the implementation a bit to have a `showRoles` prop in the main component and check if the EmbeddedChat user wants it or not. If show roles is true we could call the API and get the roles. What do you think?

**Comment by Rec0iL99** [comment]
Hey @sidmohanty11, great suggestion. I'll do that by providing a `showRoles` boolean prop in the main component to give the user freedom of whether to show roles. 

I have an idea on how we can improve our data fetching, I'll create a separate issue with the details and tag you there when it's ready!

File: src/store/rolesStore.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import create from 'zustand';
+
+const useRolesStore = create((set) => ({
+  roles: {},
```
Comment (sidmohanty11) [inline src/store/rolesStore.js:4]: I think `roles` are for "users" so I think we can have them under the userStore and not create a separate one?

File: playground/src/App.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ function App() {
           anonymousMode={true}
           headerColor="white"
           toastBarPosition="bottom-end"
+          showRoles={true}
```
Comment (sidmohanty11) [inline playground/src/App.js:24]: please update this change in the README `props`

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]
Amazing work! @Rec0iL99

File: src/store/rolesStore.js (hunk 1 above)
Comment (Rec0iL99) [reply src/store/rolesStore.js]: resolved in [52ebd6b](https://github.com/RocketChat/EmbeddedChat/pull/115/commits/52ebd6bd133397d706cfb7139afc8bac43070fb5)

File: playground/src/App.js (hunk 2 above)
Comment (Rec0iL99) [reply playground/src/App.js:24]: resolved in [52ebd6b](https://github.com/RocketChat/EmbeddedChat/pull/115/commits/52ebd6bd133397d706cfb7139afc8bac43070fb5)

File: README.md [hunk 3]
```diff
@@ -54,8 +55,7 @@ Read this [wiki page](https://github.com/RocketChat/EmbeddedChat/wiki/Roots-of-E
 | roomId           | the public room's id that you want to subscribe to                                                                                          |
 | channelName      | the fallback channel name to be present on the chat header                                                                                  |
 | anonymousMode    | if the user can see the chat without logging in                                                                                             |
-| showAvatar       | show the user's avatar in the chat component, defaults to false                                                      |
-
+| showAvatar       | show the user's avatar in the chat component, defaults to false                                                                             |
```
Comment (sidmohanty11) [inline README.md:58]: can you please add a line here on the description part?

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (Rec0iL99) [reply README.md:58]: resolved in [b3f1282](https://github.com/RocketChat/EmbeddedChat/pull/115/commits/b3f12828265f0806f03ef0edf2e5b96cd06c0c31)

**Comment by sidmohanty11** [comment]
Hey @Rec0iL99 please take a look at the merge conflicts. Thanks!

**Review by sidmohanty11 — APPROVED** [review]

**Pull request merged**